examples/step-90/step-90.cc

```diff
@@ -0,0 +1,1271 @@
+            //  triangulation.
+            // Next, the copier worker distributes the local contributions from
+            // the CopyData taking into account the constraints. Finally, the
+            // MeshWorker goes over all cells provided by the solutions'
+            // DoF-handler. Note that this includes non-intersected cells as
+            // well,
+            //  but the cell worker does nothing on them.
+            if (stabilization_scheme.needs_cell_worker())
+              {
+                typename DoFHandler<dim>::active_cell_iterator level_set_cell(
+                  &(triangulation),
+                  cell->level(),
+                  cell->index(),
+                  &level_set_dof_handler);
+                scratch_data.fe_values.reinit(cell);
+                scratch_data.level_set_fe_values.reinit(level_set_cell);
+                stabilization_scheme.assemble_cell_worker(level_set,
+                                                          cell,
+                                                          scratch_data,
+                                                          copy_data);
+              }
+          }
+      };
+
+    const auto copier = [&](const CopyData<dim> &c) {
+      constraints.distribute_local_to_global(c.cell_matrix,
+                                             c.cell_rhs,
+                                             c.local_dof_indices,
+                                             global_matrix,
+                                             global_rhs);
+    };
+
+    ScratchData<dim> scratch_data(mapping,
+                                  fe_collection,
+                                  mesh_classifier,
+                                  level_set_dof_handler,
+                                  level_set,
+                                  surface_update_flags,
+                                  cell_quadrature,
+                                  quadrature_1D);
+
+    CopyData<dim> copy_data;
+
+    MeshWorker::mesh_loop(dof_handler.begin_active(),
+                          dof_handler.end(),
+                          cell_worker,
+                          copier,
+                          scratch_data,
+                          copy_data,
+                          MeshWorker::assemble_own_cells);
+
+    global_matrix.compress(VectorOperation::add);
+    global_rhs.compress(VectorOperation::add);
+
+    timer.stop();
+    pcout << "took (" << timer.wall_time() << "s)" << std::endl;
+  }
+
+  // In the following, we solve the resulting linear system of equations. We
+  // either use a direct solver or AMG.
+  template <int dim>
+  void LaplaceBeltramiSolver<dim>::solve()
+  {
+    TimerOutput::Scope t(computing_timer, "solve");
+    Timer              timer;
+    bool               apply_direct_solver = false;
+    if (apply_direct_solver)
+      {
+        pcout << "Solving directly on MPI_rank=0... " << std::flush;
+        SolverControl solver_control(2000, 1e-8);
```

Comment:
We've been trying to use relative tolerances instead of absolute ones, see #9596.